lib/widgets/code_block.dart

```diff
@@ -1,267 +1,549 @@
+    required TextStyle sx,
+    required TextStyle sr,
+    required TextStyle s1,
+    required TextStyle ss,
+    required TextStyle bp,
+    required TextStyle vc,
+    required TextStyle vg,
+    required TextStyle vi,
+    required TextStyle il,
+  }) :
+    _hll = hll,
+    _c = c,
+    _err = err,
+    _k = k,
+    _o = o,
+    _cm = cm,
+    _cp = cp,
+    _c1 = c1,
+    _cs = cs,
+    _gd = gd,
+    _ge = ge,
+    _gr = gr,
+    _gh = gh,
+    _gi = gi,
+    _go = go,
+    _gp = gp,
+    _gs = gs,
+    _gu = gu,
+    _gt = gt,
+    _kc = kc,
+    _kd = kd,
+    _kn = kn,
+    _kp = kp,
+    _kr = kr,
+    _kt = kt,
+    _m = m,
+    _s = s,
+    _na = na,
+    _nb = nb,
+    _nc = nc,
+    _no = no,
+    _nd = nd,
+    _ni = ni,
+    _ne = ne,
+    _nf = nf,
+    _nl = nl,
+    _nn = nn,
+    _nt = nt,
+    _nv = nv,
+    _nx = nx,
+    _ow = ow,
+    _w = w,
+    _mf = mf,
+    _mh = mh,
+    _mi = mi,
+    _mo = mo,
+    _sb = sb,
+    _sc = sc,
+    _sd = sd,
+    _s2 = s2,
+    _se = se,
+    _sh = sh,
+    _si = si,
+    _sx = sx,
+    _sr = sr,
+    _s1 = s1,
+    _ss = ss,
+    _bp = bp,
+    _vc = vc,
+    _vg = vg,
+    _vi = vi,
+    _il = il;
+
+  final TextStyle plain;
+
+  final TextStyle _hll;
+  final TextStyle _c;
+  final TextStyle _err;
+  final TextStyle _k;
+  final TextStyle _o;
+  final TextStyle _cm;
+  final TextStyle _cp;
+  final TextStyle _c1;
+  final TextStyle _cs;
+  final TextStyle _gd;
+  final TextStyle _ge;
+  final TextStyle _gr;
+  final TextStyle _gh;
+  final TextStyle _gi;
+  final TextStyle _go;
+  final TextStyle _gp;
+  final TextStyle _gs;
+  final TextStyle _gu;
+  final TextStyle _gt;
+  final TextStyle _kc;
+  final TextStyle _kd;
+  final TextStyle _kn;
+  final TextStyle _kp;
+  final TextStyle _kr;
+  final TextStyle _kt;
+  final TextStyle _m;
+  final TextStyle _s;
+  final TextStyle _na;
+  final TextStyle _nb;
+  final TextStyle _nc;
+  final TextStyle _no;
+  final TextStyle _nd;
+  final TextStyle _ni;
+  final TextStyle _ne;
+  final TextStyle _nf;
+  final TextStyle _nl;
+  final TextStyle _nn;
+  final TextStyle _nt;
+  final TextStyle _nv;
+  final TextStyle _nx;
+  final TextStyle _ow;
+  final TextStyle _w;
+  final TextStyle _mf;
+  final TextStyle _mh;
+  final TextStyle _mi;
+  final TextStyle _mo;
+  final TextStyle _sb;
+  final TextStyle _sc;
+  final TextStyle _sd;
+  final TextStyle _s2;
+  final TextStyle _se;
+  final TextStyle _sh;
+  final TextStyle _si;
+  final TextStyle _sx;
+  final TextStyle _sr;
+  final TextStyle _s1;
+  final TextStyle _ss;
+  final TextStyle _bp;
+  final TextStyle _vc;
+  final TextStyle _vg;
+  final TextStyle _vi;
+  final TextStyle _il;
+
+  /// The [TextStyle] for a [CodeBlockSpanType], if there is one.
+  ///
+  /// Used to render syntax highlighting.
+  // Span styles adapted from:
+  //   https://github.com/zulip/zulip/blob/213387249e7ba7772084411b22d8cef64b135dd0/web/styles/pygments.css
+  TextStyle? span(CodeBlockSpanType type) {
```

Comment:
How about:
```suggestion
  TextStyle? forSpan(CodeBlockSpanType type) {
```

The name just "span" reads to me like it's meant to return a span.